userspace/libsinsp/container_engine/bpm.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright (C) 2013-2019 Draios Inc dba Sysdig.
+
+This file is part of sysdig.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+
+#include "container_engine/bpm.h"
+#include "sinsp.h"
+
+using namespace libsinsp::container_engine;
+
+bool bpm::resolve(sinsp_container_manager* manager, sinsp_threadinfo* tinfo, bool query_os_for_missing_info)
+{
+	sinsp_container_info container_info;
+	bool matches = false;
+
+	for(auto it = tinfo->m_cgroups.begin(); it != tinfo->m_cgroups.end(); ++it)
+	{
+		string cgroup = it->second;
+		size_t pos;
+
+		//
+		// Non-systemd and systemd BPM
+		//
+		pos = cgroup.find("bpm-");
+		if(pos != string::npos)
+		{
+			auto id_start = pos + sizeof("bpm-") - 1;
+			auto id_end = cgroup.find(".scope", id_start);
+			container_info.m_type = CT_BPM;
+			container_info.m_id = cgroup.substr(id_start, id_end - id_start);
+			matches = true;
+			break;
+		}
+	}
+
+	if (!matches)
+		return false;
+
+	tinfo->m_container_id = container_info.m_id;
+	if (!manager->container_exists(container_info.m_id))
+	{
+		container_info.m_name = container_info.m_id;
```

Comment:
Do BPM containers have any metadata except for an id?

Reply:
By metadata, do you mean [these runc annotations](https://github.com/opencontainers/runtime-spec/blob/master/config.md#annotations)?
Looking into some `config.json` as for example `/var/vcap/data/bpm/bundles/blobstore/blobstore/config.json`, I can't find `annotations`. Therefore I guess that BPM containers do not have any metadata. @xoebus please correct me if I'm wrong.

Reply:
I mean things like Docker has labels (arbitrary key-value pairs), container names, container image names (both human-readable and a sha256 hash).

Looks like e.g. the job/process names encoded in the container id could work as labels.